### PR TITLE
Add new HTML5 FAPI element: search

### PR DESCRIPTION
**Pull request by jenlampton**
PR for https://github.com/backdrop/backdrop-issues/issues/200

### CODE REVIEW

**Comment by jenlampton** [comment]
resolved merge conflicts, pushing back up.

File: core/themes/seven/style.css [hunk 1]
```diff
@@ -748,7 +750,11 @@ input.form-tel:focus,
 input.form-email:focus,
 input.form-url:focus,
 input.form-color:focus,
+<<<<<<< HEAD
 input.form-number:focus,
+=======
+input.form-search:focus,
+>>>>>>> Issue #1174628 by Dave Reid, Niklas Fiekas, ericduran, Jacine: Add new HTML5 FAPI element: search.
```
Comment (quicksketch) [inline core/themes/seven/style.css]: Oops, looks like we've got a conflict committed here. I think we can just remove the <<<, ===, and >>> lines.

**Comment by jenlampton** [comment]
cleaned up, and also rebased off 1.x

**Comment by quicksketch** [comment]
So until we make a decision on normalize.css (https://github.com/backdrop/backdrop-issues/issues/250), we'll need to include these resets in Bartik (in style.css) and Seven (in reset.css):

```
/* Reset HTML5 search elements to appear as textfields. */
input[type="search"] {
  -webkit-appearance: textfield;
  -moz-box-sizing: content-box;
  -webkit-box-sizing: content-box;
  box-sizing: content-box;
}
input[type="search"]::-webkit-search-cancel-button,
input[type="search"]::-webkit-search-decoration {
  -webkit-appearance: none;
}
```

**Comment by quicksketch** [comment]
I had to remove a small amount of CSS from Bartik that was added in this PR to be compatible with the normalize.css file added in #450. Otherwise this worked great. Thanks!

**Pull request merged**